src/table.rs

```diff
@@ -273,27 +273,65 @@ impl FatTrait for Fat16 {
 impl FatTrait for Fat32 {
     fn get_raw<T: ReadSeek>(fat: &mut T, cluster: u32) -> io::Result<u32> {
         fat.seek(io::SeekFrom::Start((cluster * 4) as u64))?;
-        Ok(fat.read_u32::<LittleEndian>()? & 0x0FFFFFFF)
+        Ok(fat.read_u32::<LittleEndian>()?)
     }
 
     fn get<T: ReadSeek>(fat: &mut T, cluster: u32) -> io::Result<FatValue> {
-        let val = Self::get_raw(fat, cluster)?;
+        let val = Self::get_raw(fat, cluster)? & 0x0FFFFFFF;
         Ok(match val {
+            0 if cluster == 0x0FFFFFF7 => {
```

Comment:
This fix is designed to avoid data corruption, if a FAT has >0x0FFFFFF7 entries.
Cluster numbers 0x0FFFFFF7 through 0x0FFFFFFF are valid on large FAT volumes.
It would be best if (most of) the file system never had to deal with this edge case.

Oh... this does need to change for the set() function, to return an Err() instead of just using warn!() macro.  I will update the PR... expect the update in a day or so...

Reply:
To answer the question as to the "n" case: Yes, because it's invalid to store a valid FAT chain that includes these special cluster numbers.

If the FAT entry for one of the clusters 0x0FFF_FFF7 through 0x0FFF_FFFF contains a value "n", that means that the FAT entry is storing a pointer to a next cluster for that value.

Reply:
OK, I've updated the code so it properly returns an error when attempting to set one of these special cluster_numbers to have a value other than BAD_SECTOR via the set() function.
See cc9a449cf52f7b0f28312ab2dffc989d660aaa34.

This also restructures the code slightly so that the block containing these errors is separate from normal non-error code flow, which should address the readability concerns.

Reply:
Can you quote specification where it says you cant have "n" in entries 0x0FFF_FFF7 through 0x0FFF_FFFF? It says it cannot be allocatable  (To avoid possible confusion by disk utilities, no FAT32 volume should ever be configured such that 0x0FFFFFF7 is an allocatable cluster number.). For me this means it can be anything except 0. Even something looking like a cluster chain or EOC marker.

Reply:
Please change those match arms into single if. It has the same logic for all added cases (warning + returning FatValue::Bad). For me it's code repetition. And I don't think different warn message justifies it. If you want to have more information in warning you can print cluster number and FAT entry value. Please do it both in get() and set() methods

Reply:
Although I disagree, **_I have now minimized arm count per your request_**.

See e802533bf96036b3af43af9104ccdc8f8fbc4e96.